### DISCUSSION TITLE
Improve documentation for Azure DevOps platforms

### DISCUSSION THREAD

**Discussion by ebisso** [renovatebot/renovate#25061] [Suggest an Idea]
### Tell us more.

While adding self-hosted renovate to our project pipelines, I have taken notes of some gap in the docs related to user permissions. I think for the benefit of users going on the same path, it would be nice if that could be included in documentation.

**Scenario**
Running self-hosted renovate in Azure DevOps (dev.azure.com)
A single pipeline is used to update multiple repositories
A dedicated user (service principal) is used by the pipeline - in other words, not using System.AccessToken.

### Documentation gap:

1. The following permissions are needed on the _repositories_:
Contribute
Contribute to pull requests
Create branch
**_Read_**

1. The following permissions are needed at _project_ level:
**_View project-level information_**

1. When using "config:azureWorkItemId", the user needs the following permissions on the _area path_ for that work item:
**_Edit work items in this node_**
**_View work items in this node_**

1. Only if you want Renovate to be able to create a tag listed in "config:addLabels" that does not exist yet, the user needs the following permissions on the _project_:
**_Create tag definition_**

**Comment by rarkins** [comment]
Source code is here. PRs welcome! https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/azure/index.md?plain=1

**Reply by ebisso** [reply]
Doc updated #25071
Thank you!
